src/backend/distributed/metadata/dependency.c

```diff
@@ -895,6 +921,55 @@ FollowAllSupportedDependencies(ObjectAddressCollector *collector,
 }
 
 
+/*
+ * FollowAllDependencies applies filters on pg_depend entries to follow the dependency
+ * tree of objects in depth first order. We will visit all objects irrespective of it is
+ * supported by Citus or not.
+ */
+static bool
+FollowAllDependencies(ObjectAddressCollector *collector,
+					  DependencyDefinition *definition)
+{
+	if (definition->mode == DependencyPgDepend)
+	{
+		/*
+		 *  For dependencies found in pg_depend:
+		 *
+		 *  Follow only normal and extension dependencies. The latter is used to reach the
+		 *  extensions, the objects that directly depend on the extension are eliminated
+		 *  during the "apply" phase.
+		 *
+		 *  Other dependencies are internal dependencies and managed by postgres.
+		 */
+		if (definition->data.pg_depend.deptype != DEPENDENCY_NORMAL &&
+			definition->data.pg_depend.deptype != DEPENDENCY_EXTENSION)
+		{
+			return false;
+		}
+	}
+
+	/* rest of the tests are to see if we want to follow the actual dependency */
+	ObjectAddress address = DependencyDefinitionObjectAddress(definition);
+
+	/*
+	 * If the object is already in our dependency list we do not have to follow any
+	 * further
+	 */
+	if (IsObjectAddressCollected(address, collector))
+	{
+		return false;
+	}
+
+	if (CitusExtensionObject(&address))
```

Comment:
I created a table that depends on  non-distributed extension:

```SQL
create table tmp(a int);
create extension seg;
alter extension seg add table tmp;
```

And, created a function that depends the table and we didn't get the warning. Do you know why? Should we have IsOwnedByExtension or such?

```
 CREATE FUNCTION fetchval_3(tmp)                                                                                      RETURNS text                                                                                                                                                   AS 'SELECT 1::text'                                                                                                                                            LANGUAGE sql STRICT;

ERROR:  type public.tmp does not exist
CONTEXT:  while executing command on localhost:9702
Time: 170.211 ms
```